### PR TITLE
Add HDF5 dependency to beluga core

### PR DESCRIPTION
**Pull request by serraramiro1**
### Proposed changes
Needed to work in #309 as our NDT generated maps are in HDF5 format. 

#### Type of change

- [ ] 🐛 Bugfix (change which fixes an issue)
- [x] 🚀 Feature (change which adds functionality)
- [ ] 📚 Documentation (change which fixes or extends documentation)

### Checklist

_Put an `x` in the boxes that apply. This is simply a reminder of what we will require before merging your code._

- [x] Lint and unit tests (if any) pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] All commits have been signed for [DCO](https://developercertificate.org/)

### CODE REVIEW

File: beluga/package.xml [hunk 1]
```diff
@@ -15,6 +15,7 @@
   <buildtool_depend>cmake</buildtool_depend>
 
   <depend>eigen</depend>
+  <depend>libhdf5-dev</depend>
```
Comment (nahueespinosa) [inline beluga/package.xml:18]: Is this not available in Focal or something? :smiling_face_with_tear:

Comment (serraramiro1) [reply beluga/package.xml:18]: It is! It seems like the old version doesn't export the `hdf5::*` variables, but we should be good to go using the other one I just switched to.
https://cmake.org/cmake/help/latest/module/FindHDF5.html

Comment (serraramiro1) [reply beluga/package.xml:18]: It is! https://packages.ubuntu.com/search?keywords=libhdf5-dev

I used another variable to link against the libraries based on:
https://cmake.org/cmake/help/latest/module/FindHDF5.html
That seems to be part of all the distributed versions.

Comment (nahueespinosa) [reply beluga/package.xml:18]: Sweet!

**Review by nahueespinosa — APPROVED** [review]
LGTM

**Pull request merged**